2018 Robot/src/org/usfirst/frc/team4999/robot/sensors/GyroFusion.java

```diff
@@ -0,0 +1,35 @@
+package org.usfirst.frc.team4999.robot.sensors;
+
+import edu.wpi.first.wpilibj.PIDSource;
+import edu.wpi.first.wpilibj.PIDSourceType;
+
+public class GyroFusion implements PIDSource {
+	VMXPi vmx = VMXPi.getInstance();
+	ADIS16448_IMU adis = new ADIS16448_IMU();
+	private PIDSourceType pidType = PIDSourceType.kRate;
+	@Override
+	public void setPIDSourceType(PIDSourceType pidSource) {
+		pidType = pidSource;
+	}
+
+	@Override
+	public PIDSourceType getPIDSourceType() {
+		return pidType;
+	}
+
+	@Override
+	public double pidGet() {
+		if(pidType == pidType.kDisplacement) {
+			 if(vmx.getTimeSinceLastPacket() > VMXPi.RESPONSECUTOFF) {
```

Comment:
Same comment as before.  Need to keep track of ADIS vs VMS angle offset so there isn't a big jump in angle if the VMX dies.

Not so important for rate since rate is absolute.